app/inpage/ts/inpage.ts

```diff
@@ -377,26 +370,16 @@ class InterceptorMessageListener {
 			})
 			if (forwardRequest.requestId !== undefined) this.outstandingRequests.get(forwardRequest.requestId)!.resolve(reply)
 		} catch (error: unknown) {
-			// if it is an Error, add context to it if context doesn't already exist
-			console.log(error)
-			console.log(messageEvent)
 			if (forwardRequest.requestId === undefined) return
-			if (error instanceof Error) {
-				const inputedErrorObject = {
-					...error,
-					code: ('code' in error && typeof error.code !== 'number') ? error.code : -32603,
-					data: !('data' in error) || error.data === undefined || error.data === null || typeof error.data !== 'object' ? { request: forwardRequest } : { ...error.data, request: forwardRequest }
-				}
-				return this.outstandingRequests.get(forwardRequest.requestId)!.reject(inputedErrorObject)
-			}
+			if (error instanceof Error) return this.outstandingRequests.get(forwardRequest.requestId)!.reject(error)
 			if (typeof error === 'object' && error !== null
 				&& 'code' in error && error.code !== undefined && typeof error.code === 'number'
 				&& 'message' in error && error.message !== undefined && typeof error.message === 'string'
 			) {
-				return this.outstandingRequests.get(forwardRequest.requestId)!.reject(new EthereumJsonRpcError(error.code, error.message, { request: forwardRequest }))
+				return this.outstandingRequests.get(forwardRequest.requestId)!.reject(new EthereumJsonRpcError(error.code, error.message))
```

Comment:
Same comment as above, null assertion operator should only be used in places where the compiler fails to correctly analyze the code.  Otherwise, we should handle undefined.
```suggestion
				return this.outstandingRequests.get(forwardRequest.requestId).reject(new EthereumJsonRpcError(error.code, error.message))
```